### DISCUSSION TITLE
Get a video direction on custom module

### DISCUSSION THREAD

**Discussion by GOBIKRISHNAN-R** [baresip/baresip#1831] [Q&A]
Dear Folks,

I'm trying to get the video direction on my own custom module. I'm expecting to print the sdp_stream_video_direction on my module.

How to get the video stream direction values in the custom module on baresip?
Any suggestions?

Thakks
Gobi

**Comment by sreimers** [comment]
If needed we can move `stream_ldir` from core.h to baresip.h and you can maybe use 
```c 
stream_ldir(video_strm(call_video(call)));
``` 

But if you only need this for debugging you can also use `stream_debug()` already.

**Reply by cspiel1** [reply]
You can use the UA events. Every call related UA event contains a JSON which has fields `videodir`.

Thus you need an ua-event handler.
